### ISSUE TITLE
Improve the way we teach elementary workflow

### ISSUE THREAD

**Issue by kaixoo** [elementary/switchboard-plug-applications#247]
### Problem

Elementary humanistic guidelines express:

> Because of the strong convention of saved state, elementary OS does not expose or optimize for legacy minimize behavior; e.g. there is no minimize button, and the Multitasking View does not distinguish minimized windows.

However, if you don't know this, you can keep using Ctrl+H forever.

### Proposal

For minimizing: show an informative dialog when the user uses Ctrl+H (in an elementary-curated app):

> Hey! In elementary applications, closing an application saves all of its state: it equals minimizing in other operating systems.

> Do not show more  |  Ignore

**Comment by jeremypw** [comment]
> Hey! In elementary applications, closing an application saves all of its state: it equals minimizing in other operating systems.

Save-state-on-closing, while laudable, is not equal to minimizing in all respects because any ongoing processes are terminated and are not resumed on re-opening (yet).  If you want to e.g. continue a lengthy search or file operation you have to move the window to another workspace to get it out the way.

**Comment by kaixoo** [comment]
This is not that:

> Apps should save their current state when closed so they can be reopened right to where the user left off. **Typically, closing and reopening an app should be indistinguishable from the legacy concept of minimizing and unminimizing an app**

> Apps should never minimize instead of closing, as that puts the app window into a state that is foreign to users of elementary OS. Instead, windows should close or hide and re-open with a [saved state](https://docs.elementary.io/hig/user-workflow#saving-state).

By that same philosophy, Files or Web app should keep ongoing operations when closed.

**Comment by jeremypw** [comment]
> By that same philosophy, Files or Web app should keep ongoing operations when closed.

Absolutely, but we are not there yet as it requires a lot of reworking.  The relevant operations have to be split off into separate background processes that can be run independently of the UI.

**Comment by kaixoo** [comment]
Okay, but don't you think we should have a way to express that minimizing or doing `Ctrl+H` is not the *elementary way* from within elementary?
